Khan/wonder-blocks
# [feature:dropdown] WB-632 - Add controlled mode to ActionMenu, SingleSelect and MultiSelect

## Summary
Adds a "controlled" mode to all the dropdown variants. To make this mode work, you need to add the `opened` and `onToggle` props to the component instance. The dropdown openers still work on this mode due to the dropdown/select nature.

## Features

### Dropdown
- Changed `onClose` to `onToggle` for better understanding on the controlled mode. This means that `onToggle` will be triggered every time the menu is open or closed.

### ActionMenu, SingleSelect, MultiSelect
- Added `getDerivedStateFromProps` to sync the `opened` prop with the parent's state.
- Added `onToggle` prop to sync up the `opened` prop within the parent component.
- Updated unit tests.
- Added examples.

### Test plan
1. Open https://deploy-preview-457--wonder-blocks.netlify.com/#section-dropdown
2. Check the documentation is clear enough.
3. Go to the examples related to controlled components and check that everything works as expected:
    - ActionMenu: https://deploy-preview-457--wonder-blocks.netlify.com/#!/ActionMenu/11
    - SingleSelect: https://deploy-preview-457--wonder-blocks.netlify.com/#!/SingleSelect/15
    - MultiSelect: https://deploy-preview-457--wonder-blocks.netlify.com/#!/MultiSelect/15
